packages/shortcuts-extension/src/components/ShortcutList.tsx

```diff
@@ -23,10 +25,91 @@ export interface IShortcutListProps {
   height: number;
   contextMenu: Function;
   external: IShortcutUIexternal;
+  id?: string;
 }
 
 /** React component for list of shortcuts */
 export class ShortcutList extends React.Component<IShortcutListProps> {
+  handleTabIndex(shortcut: ShortcutObject): number {
+    let title = shortcut.commandName;
+    let firstElement = this.props.shortcuts[0].commandName;
+
+    if (firstElement === title) {
+      return 0;
+    }
+    return -1;
+  }
+
+  /**
+   * Handle key down for row navigation
+   */
+  handleRowKeyDown = (event: React.KeyboardEvent): void => {
+    if (ARROW_KEYS.includes(event.key)) {
+      let shortcutList;
+
+      if (this.props.id) {
+        shortcutList = document.getElementById(this.props.id) as HTMLElement;
+      }
+
+      const focusable: Element[] = [];
+
+      if (shortcutList) {
+        // Get focusable children within the shortcut list
+        Array.from(shortcutList.children).forEach(child => {
+          focusable.push(child);
+        });
+        console.log(`Array length: ${focusable.length}`);
+        // If focusable contains only one element, nothing to do.
+        if (focusable.length <= 1) {
+          return;
+        }
+        event.preventDefault();
+        event.stopPropagation();
+      }
+      let activeElement = document.activeElement as HTMLElement;
+      let currentNode = focusable.indexOf(activeElement);
+      let activeNode = focusable[currentNode] as HTMLElement;
+      let nextNode = focusable[currentNode + 1] as HTMLElement;
+      let previousNode = focusable[currentNode - 1] as HTMLElement;
+
+      if (event.key === 'ArrowDown') {
+        let nxtNode = nextNode;
+        if (nxtNode) {
+          nxtNode.setAttribute('tabindex', '0');
+          activeNode.setAttribute('tabindex', '-1');
```

Comment:
This line throws an error when the function is called by ShortcutItem. That's because the active element in that case is not one of the direct children of the ShortcutList element, so `activeNode` ends up undefined.

Reply:
Hi @gabalafou,

Just working through these comments and after removing things like arrow nav wrapping, extra key down handler etc. I'm unable to replicate this error